mesonbuild/scripts/externalproject.py

```diff
@@ -48,27 +48,23 @@ def write_stampfile(self) -> None:
         with open(self.stampfile, 'w', encoding='utf-8'):
             pass
 
-    def gnu_make(self) -> bool:
+    def supports_jobs_flag(self) -> bool:
         p, o, e = Popen_safe(self.make + ['--version'])
-        if p.returncode == 0 and 'GNU Make' in o:
+        if p.returncode == 0 and ('GNU Make' in o or 'waf' in o):
             return True
         return False
 
     def build(self) -> int:
-        is_make = self.make[0] == 'make'
         make_cmd = self.make.copy()
-        if is_make and self.gnu_make():
+        if self.supports_jobs_flag():
             make_cmd.append(f'-j{multiprocessing.cpu_count()}')
         rc = self._run('build', make_cmd)
         if rc != 0:
             return rc
 
         install_cmd = self.make.copy()
         install_env = {}
-        if is_make:
-            install_cmd.append(f'DESTDIR={self.install_dir}')
-        else:
-            install_env['DESTDIR'] = self.install_dir
+        install_env['DESTDIR'] = self.install_dir
```

Comment:
I'm not so sure about this change, IIRC the reason I made it like that for make is because it does not always work with DESTDIR in env instead of in args.